### PR TITLE
allow git@github.com in extractGithub. Fixes #1549

### LINKED ISSUE #1549: Support git@...:... repository URLs
The [extractGithub](https://github.com/purescript/purescript/blob/3794bda/src/Language/PureScript/Publish.hs#L195) function requires URLs to start with `git://github.com/`. The clone URLs provided by GitHub start with `git@github.com:`, so they are rejected.

### PR DESCRIPTION
**Pull request by codedmart**
Didn't see anywhere that the url is tested. Please let me know if I missed anything. Just starting to look around the codebase.

### CODE REVIEW

**Comment by phadej** [comment]
I'd almost use `regex-applicative` or `parsec` here. The ad-hoc parsing is getting out-of-hands.

**Comment by paf31** [comment]
Looks good, but I agree with @phadej that a parsing library might be a better fit now.

Could you please update `CONTRIBUTORS.md` too?

**Comment by codedmart** [comment]
@paf31 OK updated `CONTRIBUTORS.md` should I make the change to a parsing library now? Or should we leave this and add another issue for that. Are there other areas we could use the parser throughout the codebase? Thanks!

File: src/Language/PureScript/Publish.hs [hunk 1]
```diff
@@ -212,6 +212,11 @@ extractGithub =
     | ".git" `isSuffixOf` str = take (length str - 4) str
     | otherwise = str
 
+  matchUrl :: String -> Maybe String
+  matchUrl str
+    | "git@github.com:" `isPrefixOf` str = stripPrefix "git@github.com:" str
+    | otherwise = stripPrefix "git://github.com/" str
+
```
Comment (hdgarrood) [inline src/Language/PureScript/Publish.hs]: Would `stripPrefix "git@github.com:" str <|> stripPrefix "git://github.com/" str` work here?

Comment (hdgarrood) [inline src/Language/PureScript/Publish.hs]: Although I guess it doesn't matter if you're about to switch to parsec or something anyway. I am fairly sure this is the only part of the codebase that deals with these kinds of URLs, too.

**Comment by paf31** [comment]
I'll leave this up to @hdgarrood to decide if it's worth switching to a parser library or not.

**Comment by hdgarrood** [comment]
I'm generally :-1: for regexes in real code (as opposed to, say, a shell pipeline command that you might run as a one-off). Yes, this is a little ad-hoc, but so are regexes, and this kind of code is much easier to read or modify.

`parsec` seems reasonable, but I'm not sure it's worth it here, especially since the chance of what we need to parse here becoming any more complex seems rather slim.

**Comment by paf31** [comment]
I think `regex-applicative` makes things a little less ad-hoc, but you still get the speed benefits. Not that speed is a particular concern here.

**Comment by codedmart** [comment]
@hdgarrood What would you like me to do? Fix this up with something like `regex-applicative` or leave it as is or make the other change you suggested `stripPrefix "git@github.com:" str <|> stripPrefix "git://github.com/" str`?

**Comment by hdgarrood** [comment]
Now that I've had a proper look at regex-applicative it doesn't look like too bad an option. When I said :-1: I was imagining a single regex that would extract the GitHub username and password through capturing groups.

Anyway I still think sticking with the current approach is ok for now, but I could be persuaded otherwise. If we do stick with the current approach then i do think <|> would be better. Using isPrefixOf as well as stripPrefix in the same line is a code smell, I think.

**Comment by codedmart** [comment]
Fair enough I will make some changes. Thanks!

**Comment by paf31** [comment]
I'm fine with keeping the current approach, but I agree that `<|>` would make the code easier to read.

**Comment by codedmart** [comment]
Sorry guys took a break for a few days. I will get these changes done this Monday. Thanks!

**Comment by paf31** [comment]
No rush :smile:

**Comment by codedmart** [comment]
Changes made. Let me know if there is anything else.

**Comment by paf31** [comment]
Looks good to me, thanks! Could you please squash your commits?

**Comment by codedmart** [comment]
Done

**Comment by paf31** [comment]
@hdgarrood Is this ok now?

**Comment by hdgarrood** [comment]
Looks good to me :+1:

**Comment by paf31** [comment]
Thanks!

**Comment by tfausak** [comment]
:+1: Thanks!

**Pull request merged**